src/lib.rs

```diff
@@ -116,6 +116,7 @@ pub use time::{Time, IndexTime};
 pub use tree::{Tree, TreeEntry, TreeIter};
 pub use treebuilder::TreeBuilder;
 pub use util::IntoCString;
+pub use filter::{StreamFilter, WriteStream, FilterChain, RawStream, FilterSource};
```

Comment:
API-wise, maybe this could work like transports where `filter` is just a public module with exports inside of it?